File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.java
class NumArray {
    int numarray[];
    public NumArray(int[] nums) {
        this.numarray=nums;
    }
    
    public int sumRange(int left, int right) {
         int total=0;
        for(int i=left;i<=right;i++)
        {
            total+=numarray[i];
        }
        return total;
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.cpp
class NumArray {
public:
    vector<int> a,presum;
    NumArray(vector<int>& nums) {
        a=nums;
        presum.resize(a.size());
        presum[0]=a[0];
        for(int i=1;i<a.size();i++){
            presum[i]=a[i]+presum[i-1];
        }
    }
    
    int sumRange(int left, int right) {
        if(left==0)
            return presum[right];
        return presum[right]-presum[left-1];
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */